### PR TITLE
Maintenance/rename ginkgo dpcpp to sycl

### PR DESCRIPTION
**Pull request by balos1**
Ginkgo now refers to DPC++ as SYCL (inline with what Intel is doing), so this just updates our interface to match the naming convention.

### CODE REVIEW

**Comment by balos1** [comment]
Note that we will need to update our Spack package to handle this change.

File: doc/shared/RecentChanges.rst [hunk 1]
```diff
@@ -79,3 +79,5 @@ Fix bug on LLP64 platforms (like Windows 64-bit) where ``KLU_INDEXTYPE`` could b
 
 Check if size of ``SuiteSparse_long`` is 8 if the size of ``sunindextype`` is 8
 when using KLU.
+
+Renamed the DPCPP value for the :cmakeop:`SUNDIALS_GINKGO_BACKENDS` CMake option to `SYCL` to match Ginkgo's updated naming convention.
```
Comment (gardner48) [inline doc/shared/RecentChanges.rst:83]:
```suggestion
Renamed the DPCPP value for the :cmakeop:`SUNDIALS_GINKGO_BACKENDS` CMake option to ``SYCL`` to match Ginkgo's updated naming convention.
```

File: doc/shared/sundials/Install.rst [hunk 2]
```diff
@@ -665,7 +665,7 @@ illustration only.
 .. cmakeoption:: SUNDIALS_GINKGO_BACKENDS
 
    Semi-colon separated list of Ginkgo target architecutres/executors to build for.
-   Options currenty supported are REF (the Ginkgo reference executor), OMP, CUDA, HIP, and DPC++.
+   Options currenty supported are REF (the Ginkgo reference executor), OMP, CUDA, HIP, and SYCL.
 
    Default: "REF;OMP"
```
Comment (gardner48) [inline doc/shared/sundials/Install.rst:670]:
```suggestion
   Default: "REF;OMP"

   .. versionchanged: x.y.z 

      The ``DPCPP`` option was changed to ``SYCL`` to align with Ginkgo's naming convention.
```

**Review by gardner48 — CHANGES_REQUESTED** [review]
Minor doc items, otherwise this looks good

**Review by drreynolds — APPROVED** [review]

**Review by gardner48 — APPROVED** [review]

**Review by gardner48 — APPROVED** [review]

**Pull request merged**